### PR TITLE
Product detail part implementation

### PR DESCRIPTION
**Pull request by DarinaBlazevica**
-Product detail component added(styling, content)
-Lazy loading for component
-Dialog box added
-Redirection to Dialog Box buttons added
-Cart component added for redirection (full implementation in another PR)

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -2,23 +2,32 @@ import { NgModule } from '@angular/core';
...
 const routes: Routes = [
+  {
+    path: 'main',
+    loadChildren: () =>
+      import('./components/main-page/main-page.module').then(
+        (m) => m.MainPageModule
+      ),
+  },
+  {
+    path: 'bookDetails/:id',
```
Comment (Jalal999) [inline src/app/app-routing.module.ts]: books/:id or products/:id would be more appropriate name for path.

File: src/app/components/book-details/book-details.component.css [hunk 2]
```diff
@@ -1,7 +1,52 @@
-input[type=number]::-webkit-inner-spin-button {
-    cursor: pointer;
+.book_details__container {
+    display: flex;
+    align-items: center;
+    flex-direction: column;
+    margin-top: 55px;
+    padding: 10px 10px 10px 10px;
```
Comment (Jalal999) [inline src/app/components/book-details/book-details.component.css]: padding: 10px;

File: src/app/components/book-details/book-details.component.css [hunk 3]
```diff
@@ -1,7 +1,52 @@
...
+
+.book_details__add_to_cart {
+    width: 100%;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+}
+
+.book_details__add_to_cart p {
+    padding: 10px 25px 10px 10px
```
Comment (Jalal999) [inline src/app/components/book-details/book-details.component.css]: ; at the end of line

File: src/app/components/dialog-box/dialog-box.component.css [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    right: 0;
+    background-color: #000;
+    opacity: 0.4;
+    z-index: 1;
+}
+
+
+.dialogBox button {
+    background-color: #fff9ef;
+    padding: 10px 10px 10px 10px;
```
Comment (Jalal999) [inline src/app/components/dialog-box/dialog-box.component.css]: padding: 10px;

File: src/app/components/dialog-box/dialog-box.component.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  @Input() closed = true;
+  @Input() opened!: boolean;
+  @Output() openedChange = new EventEmitter<boolean>();
+
+  constructor() {}
+
+  public toCart = 'Go to Cart';
+  public continueShop = 'Continue shopping';
+
+  ngOnInit(): void {}
```
Comment (Jalal999) [inline src/app/components/dialog-box/dialog-box.component.ts]: if you won't need ngOnInit and constructor, remove them

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (DarinaBlazevica) [reply src/app/app-routing.module.ts]: changed

File: src/app/components/book-details/book-details.component.css (hunk 2 above)
Comment (DarinaBlazevica) [reply src/app/components/book-details/book-details.component.css]: Changed

File: src/app/components/book-details/book-details.component.css (hunk 3 above)
Comment (DarinaBlazevica) [reply src/app/components/book-details/book-details.component.css]: Changed

File: src/app/components/dialog-box/dialog-box.component.css (hunk 4 above)
Comment (DarinaBlazevica) [reply src/app/components/dialog-box/dialog-box.component.css]: Changed

File: src/app/components/dialog-box/dialog-box.component.ts (hunk 5 above)
Comment (DarinaBlazevica) [reply src/app/components/dialog-box/dialog-box.component.ts]: Deleted

**Review by Jalal999 — APPROVED** [review]

**Review by ilze-dom — APPROVED** [review]

**Review by VSlobolinskyi — APPROVED** [review]

**Pull request merged**